eng/pipelines/release-ext-demo.yml

```diff
@@ -0,0 +1,33 @@
+# Continuous deployment trigger
```

Comment:
Nit: We may want to come up with naming strategy for these pipelines based off the "id" of the extension.  For example, the full id of this extension is `microsoft.azd.demo` so maybe the pipeline should be `release-ext-microsoft-azd-demo.yml` ?